src/js/components/Select/Select.js

```diff
@@ -158,11 +158,23 @@ const Select = forwardRef(
         setValue(nextValue);
         if (onChange) {
           event.persist();
-          const adjustedEvent = event;
-          adjustedEvent.target = inputRef.current;
-          adjustedEvent.value = nextValue;
-          adjustedEvent.option = option;
-          adjustedEvent.selected = nextSelected;
+          let adjustedEvent;
+          // support for native event used by Preact
+          if (event instanceof Event) {
```

Comment:
Another option is to have this condition `event.constructor.name !== 'SyntheticBaseEvent'`

Reply:
Would it be reasonable to use the `new event.constructor` approach in all cases?